lib/recent.go

```diff
@@ -0,0 +1,100 @@
+package lib
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type RecentFile struct {
+	Terraform []string
+	OpenTofu  []string
+}
+
+func appendRecentVersionToList(versions []string, requestedVersion string) []string {
+	// Check for requestedVersion in versions list
+	for versionIndex, versionVal := range versions {
+		if versionVal == requestedVersion {
+			versions = append(versions[:versionIndex], versions[versionIndex+1:]...)
+		}
+	}
+
+	// Add new version to start of slice
+	versions = append([]string{requestedVersion}, versions...)
+	if len(versions) > 3 {
```

Comment:
From https://github.com/warrensbox/terraform-switcher/pull/436#discussion_r1615278360
@yermulnik:

This pushed me to think that we may allow user to customize number of recent versions we keep for them. WDYT?

Reply:
Instead of letting users customize, lets leave it at 5 for now (increase from 3). I don't think we should over-engineer this. Unless multiple people opens a PR for this.

Reply:
This has been replaced by the work in #437